src/mpi/comm/comm_split.c

```diff
@@ -477,10 +478,8 @@ int MPI_Comm_split(MPI_Comm comm, int color, int key, MPI_Comm *newcomm)
     
     mpi_errno = MPIR_Comm_split_impl(comm_ptr, color, key, &newcomm_ptr);
     if (mpi_errno) MPIR_ERR_POP(mpi_errno);
-    if (newcomm_ptr)
```

Comment:
This change also looks unrelated to the commit it is in, but I also think it is incorrect. `MPIR_Comm_split_impl` will set `newcomm_ptr` to `NULL` in the case where a process supplies `MPI_UNDEFINED` in the color argument. The user code will then get `MPI_COMM_NULL` as the returned handle.